### ISSUE TITLE
flexible whip effect check

### ISSUE THREAD

**Issue by Polaritz** [bluemagic123/tModLoader#4148]
### Do you intend to personally contribute/program this feature?

No

### I would like to see this change made to improve my experience with

Mod capability as a Modder

### Description

a new public attribute that projectiles inherit that checks whether the projectile grants whip effect

### What does this proposal attempt to solve or improve?

as far as i checked, whip effects checks 2 conditions; 
```Projectile.minion == true``` or ```ProjectileID.Sets.Minionshot[] == true```
if modded minion uses preexisting projectile from vanilla or modded one that do not include those conditions(e.g. any bullet), it is not possible to manually apply whip effect on them

### Which (other) solutions should be considered?

Add a condition like ```bool Projectile.grantsWhipEffect``` or something to track whip effect in real time

**Comment by ghost** [comment]
So, if i am reading this correct, you want the whip the make an effect in real time?

**Comment by Polaritz** [comment]
> So, if i am reading this correct, you want the whip the make an effect in real time?

yes, by adding an public attribute that checks whip effect, like how modder easily changes pre-existing projectile's damage class by using ```Main.projectile[p].DamageType = ...```
